calblueprint/girls-write-now
# [auth] Implement forgot password & reset password

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

Adds 
- forgot password link on the log-in page
    - redirects to a forgot password screen where the user enters their email and verification code, then sets a new password
- change password button on the settings page.
    - redirects to a page where the user can change their password

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links

None

### Notion Sprint Task

[Notion task](https://www.notion.so/calblueprint/auth-Implement-forgot-password-reset-password-9cf7371bf9be4c8e9eb78842cb21eec9?pvs=4)

[//]: # 'Add the relevant Notion link(s) here'

### Online sources
None

### Related PRs

None

## How to review

- Create a new account and go through the onboarding process. You can use https://10minutemail.net/?lang=en to create an email.
- Go to the settings page and click on the change password button and try changing the password to the current password. This should not work.
- Try changing the password to something different. Log out and try the old password. Then try the new password to sign in.
- Log out from the setting page and click on the forgot password button
- Follow the on-screen directions to reset your password.
- Log out and try the new password to sign in

## Next steps

None

## Tests Performed, Edge Cases
See "How to review"

### Screenshots
![IMG_0F505B8A1414-1](https://github.com/calblueprint/girls-write-now/assets/34043950/788a52ba-3b27-44a9-a6c2-aa6fbcd3a1e9)
ts


CC: @akshaynthakur
